source/features/remove-is-open-is-closed.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+
+const countMatches = (string: string, regex: RegExp): number => {
+	return ((string || '').match(regex) ?? []).length;
+};
+
+function createMergeLink(): HTMLAnchorElement {
+	const linkMergedSearchParameters = new URLSearchParams(location.search);
+	const linkIsMerged: HTMLAnchorElement = (
+		<a href="" className="btn-link">
+		Merged
+		</a>
+	);
+	const regexpQueryTotal = /is:open|is:closed|is:issue/g;
+	const regexpQuery = /is:open|is:closed|is:issue/;
+
+	let linkMergedSearchString = new URLSearchParams(location.search).get('q') ?? select<HTMLInputElement>('#js-issues-search')!.value;
+
+	while (countMatches(linkMergedSearchString, regexpQueryTotal) > 1) {
+		linkMergedSearchString = linkMergedSearchString.replace(regexpQuery, '').trim();
+	}
+
+	if (countMatches(linkMergedSearchString, /is:merged/) === 1) {
+		linkMergedSearchParameters.set('q', linkMergedSearchString.replace(/is:merged/, 'is:issue').trim());
+		linkIsMerged.classList.add('selected');
+	} else if (countMatches(linkMergedSearchString, regexpQueryTotal) === 1) {
+		linkMergedSearchParameters.set('q', linkMergedSearchString.replace(regexpQuery, 'is:merged').trim());
+	}
+
+	linkIsMerged.search = String(linkMergedSearchParameters);
```

Comment:
To generate this link, try:
- cloning the “closed” link: cloneNode(true)
- replacing the `textContent` to Merged
- running this on its query: `.replace('is:closed', 'is:merged')`

Reply:
Nice, this way I would not get anymore the typescript error telling me that there are some attributes missing.

However, I doubt that your third point may work, because if the **closed** filter is ON, it's query would not have the `is:closed` filter, so the `.replace('is:closed', 'is:merged')` will not work. That's why I prefer the way it is.

Or maybe I must choose the **open** link to clone if the **closed** link is selected, and vice versa. This way, i will always have the `/is:closed|is:open/` string to replace by is:merged... I'm gonna give it a try!

Reply:
I've done it. Before calling the function to create the merge link, I check which link I have available on the current page: open, closed or total, and which one is selected. After this check, I clone the appropiate link using cloneNode and send it to the function so it can make the right search query. The fix is uploaded.

Reply:
Always prefer less code, that's the point of using `cloneNode` instead of creating a new link. All those conditions aren't readable.

Also I suggested the function name `addMergeLink` because it should literally add the merge link, on its own, without external help. It's easier to reason about functions when they're self-contained.

Refer to 872cd8a7